pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -278,5 +287,37 @@ private boolean isPartitionMatch(FilterQueryTree filterQueryTree, PartitionInfo
       _partitionFunction = partitionFunction;
       _partitions = partitions;
     }
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      PartitionInfo that = (PartitionInfo) o;
+      return Objects.equals(_partitionFunction, that._partitionFunction) &&
+        Objects.equals(_partitions, that._partitions);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(_partitionFunction, _partitions);
+    }
+  }
+
+  /**
+   * This class is created to speed up partition number calls. Sometimes with lots of segments (like 40k+)
+   * Computing partition values may be CPU intensive, caching the computation result may help
+   */
+  private static class CachedPartitionFunction {
+    final Map<PartitionFunction, Map<String, Integer>> cache = new HashMap<>();
```

Comment:
It would be great to add a brief javadoc on the meaning of the inner hash map.

Reply:
Added Java doc